### PR TITLE
Add command-line option to disable logs (closes #48)

### LINKED ISSUE #48: How to remove log messages
Hi

I want to use btfs on a Live distribution (Debian) : all is on memoy.

Currently, i read files stored on a nfs server. 
Now, i want to use a bittorrent server.

But, btfs does very verbose logs : with 5 minutes, all the memory is full

With the help of the doc of libtorrent, i tried to disable the logs : in the file btfs.cc, i replaced
```
                libtorrent::alert::tracker_notification |
                libtorrent::alert::stats_notification |  
                libtorrent::alert::storage_notification |
                libtorrent::alert::progress_notification |
                libtorrent::alert::status_notification | 
                libtorrent::alert::error_notification | 
                libtorrent::alert::dht_notification |
                libtorrent::alert::peer_notification;
```
by
`0;`

The program builds and runs. But, the files are not shared.

How can i remove these very verbose logs ?

Thanks.

### PR DESCRIPTION
**Pull request by DrDub**
Set the logger to `/dev/null` when run with `-s` or `--silent`.

### CODE REVIEW

File: src/btfs.cc [hunk 1]
```diff
@@ -992,6 +995,9 @@ main(int argc, char *argv[]) {
 	if (fuse_opt_parse(&args, &params, btfs_opts, btfs_process_arg))
 		RETV(fprintf(stderr, "Failed to parse options\n"), -1);
 
+        if (params.silent)
+                printf("No logs.\n");
```
Comment (johang) [inline src/btfs.cc:999]: I don't like this print. Silent should be no prints.

**Comment by johang** [comment]
Can you add `--slient` to the man page (`man/btfs.1`) too?

File: src/btfs.h [hunk 2]
```diff
@@ -100,7 +100,7 @@ class Array
 class Log : public std::ofstream
 {
 public:
-	Log(std::string p) : std::ofstream(p.c_str()), path(p) {
+       Log(std::string p) : std::ofstream(p.empty() ? "/dev/null" : p.c_str()), path(p) {
```
Comment (johang) [inline src/btfs.h:103]: Indent with tab here like rest of the code.

File: src/btfs.cc [hunk 3]
```diff
@@ -663,7 +663,7 @@ btfs_init(struct fuse_conn_info *conn) {
 #endif
 
 	pthread_create(&alert_thread, NULL, alert_queue_loop,
-		new Log(p->save_path + "/../log.txt"));
+               new Log(params.silent ? std::string() : (p->save_path + "/../log.txt")));
```
Comment (johang) [inline src/btfs.cc:666]: Indent with tab here like rest of the code.

**Review by johang — CHANGES_REQUESTED** [review]

File: src/btfs.cc (hunk 1 above)
Comment (DrDub) [reply src/btfs.cc:999]: Will drop it

File: src/btfs.h (hunk 2 above)
Comment (DrDub) [reply src/btfs.h:103]: Sure. Missed that on my editor.

**Comment by johang** [comment]
Thanks!

**Pull request merged**